crates/matrix-sdk/src/authentication/oidc/tests.rs

```diff
@@ -450,18 +449,24 @@ async fn test_insecure_clients() -> anyhow::Result<()> {
         // Restore the previous session so we have an existing set of refresh tokens.
         oidc.restore_session(mock_session(prev_tokens.clone(), server_url.clone())).await?;
 
-        let mut session_token_stream = oidc.session_tokens_stream().expect("stream available");
-
-        assert_pending!(session_token_stream);
+        let mut session_changes = client.subscribe_to_session_changes();
 
         // A refresh in insecure mode should work Just Fine.
         oidc.refresh_access_token().await?;
 
-        assert_next_matches!(session_token_stream, new_tokens => {
-            assert_eq!(new_tokens, next_tokens);
-        });
-
-        assert_pending!(session_token_stream);
+        assert_eq!(client.session_tokens().unwrap(), next_tokens);
+
+        // We get notified once that the tokens were refreshed.
+        assert_eq!(
+            session_changes.try_recv(),
+            Ok(SessionChange::TokensRefreshed),
+            "The session changes should be notified of the tokens refresh"
+        );
+        assert_eq!(
+            session_changes.try_recv(),
+            Err(TryRecvError::Empty),
+            "There should be no more session changes"
+        );
```

Comment:
I think you can use `session_changes.is_empty()` too in this case, but that's fine, just a side little note.